applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.cpp

```diff
@@ -0,0 +1,1319 @@
+    DofsVectorType& rElementalDofList,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    const auto& r_geometry     = GetGeometry();
+    const SizeType n_nodes     = r_geometry.PointsNumber();
+    const SizeType dim         = r_geometry.WorkingSpaceDimension();
+    const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+    const SizeType block_size  = dim + strain_size;
+    const SizeType dof_size    = n_nodes * block_size;
+    const SizeType u_size      = dim * n_nodes;
+
+    if (rElementalDofList.size() != dof_size) {
+        rElementalDofList.resize(dof_size);
+    }
+
+    if (dim == 2) {
+        for(IndexType i_node = 0; i_node < n_nodes; ++i_node) {
+            const auto &r_node = GetGeometry()[i_node];
+            const SizeType u_block = i_node * dim;
+            const SizeType E_block = i_node * strain_size + u_size;
+            rElementalDofList[u_block    ] = r_node.pGetDof(DISPLACEMENT_X);
+            rElementalDofList[u_block + 1] = r_node.pGetDof(DISPLACEMENT_Y);
+            rElementalDofList[E_block    ] = r_node.pGetDof(NODAL_STRAIN_VECTOR_XX);
+            rElementalDofList[E_block + 1] = r_node.pGetDof(NODAL_STRAIN_VECTOR_YY);
+            rElementalDofList[E_block + 2] = r_node.pGetDof(NODAL_STRAIN_VECTOR_XY);
+        }
+    } else if (dim == 3) {
+        for(IndexType i_node = 0; i_node < n_nodes; ++i_node) {
+            const auto &r_node = GetGeometry()[i_node];
+            const SizeType u_block = i_node * dim;
+            const SizeType E_block = i_node * strain_size + u_size;
+            rElementalDofList[u_block    ] = r_node.pGetDof(DISPLACEMENT_X);
+            rElementalDofList[u_block + 1] = r_node.pGetDof(DISPLACEMENT_Y);
+            rElementalDofList[u_block + 2] = r_node.pGetDof(DISPLACEMENT_Z);
+            rElementalDofList[E_block    ] = r_node.pGetDof(NODAL_STRAIN_VECTOR_XX);
+            rElementalDofList[E_block + 1] = r_node.pGetDof(NODAL_STRAIN_VECTOR_YY);
+            rElementalDofList[E_block + 2] = r_node.pGetDof(NODAL_STRAIN_VECTOR_ZZ);
+            rElementalDofList[E_block + 3] = r_node.pGetDof(NODAL_STRAIN_VECTOR_XY);
+            rElementalDofList[E_block + 4] = r_node.pGetDof(NODAL_STRAIN_VECTOR_YZ);
+            rElementalDofList[E_block + 5] = r_node.pGetDof(NODAL_STRAIN_VECTOR_XZ);
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SmallDisplacementMixedStrainDisplacementElement::Initialize(
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Initialization should not be done again in a restart!
+    if (!rCurrentProcessInfo[IS_RESTARTED]) {
+        // Integration method initialization
+        mThisIntegrationMethod = GeometryData::IntegrationMethod::GI_LOBATTO_1;
```

Comment:
Maybe we can do this definable from outside and, if not provided, use `GI_LOBATTO_1`. Based on our previous tests, it might be handy.

Reply:
(See my comment below)